src/file_pipe_log/pipe.rs

```diff
@@ -411,16 +432,11 @@ impl<F: FileSystem> SinglePipe<F> {
         let purged_len = purged_files.len();
         if purged_len > 0 {
             let remains_capacity = self.capacity.saturating_sub(len);
-            let (recycled_start, mut recycled_len) = {
-                let files = self.recycled_files.read();
-                files
-                    .front()
-                    .map_or_else(|| (DEFAULT_FIRST_FILE_SEQ, 0), |f| (f.seq, files.len()))
-            };
+            let mut recycled_len = self.recycled_files.read().len();
             let mut new_recycled = VecDeque::new();
-            // The newly purged files from `self.active_files` should be renamed
-            // to recycled files with LOG_APPEND_RESERVED_SUFFIX suffix, to reduce the
-            // unnecessary recovery timecost when restarting.
+            // We don't rename the append file because on some platform it could cause I/O
+            // jitters. Instead we best-effort delete them during shutdown to reduce
```

Comment:
Once best-effort flush memtable is implemented, not sure if this will slow down the flush memtable. 
Also in tikv restart, the recreate the file takes extra IOs.  To me delete files seems not the efficient way. Can we do the rename in the start? It will be more efficient than recreate anyway.